test/test-viz.cpp

```diff
@@ -0,0 +1,311 @@
+    virtual void on_input(const std::string& name) const override {
+        scheme.add_input(name);
+    }
+
+    virtual void on_output(const std::string& name) const override {
+        scheme.add_output(name);
+    }
+
+    virtual void on_dff(const std::string &input, const std::string &output) const override {
+        scheme.add_dff(input, output);
+    }
+
+    virtual void on_gate(const std::vector<std::string>& inputs, const std::string& output, const std::string& type) const override {
+        scheme.add_gate(inputs, output, type);
+    }
+
+    logic_scheme& scheme;
+};
+
+void setPosition(std::vector<Vertex*> elements, const int max_elements, const float height, const float width, const int screen_height) {
+    int count = 0;
+    int freeSpaces = elements.size() + 1;
+    float step = (screen_height - height * elements.size()) / freeSpaces;
+    for(Vertex* elem : elements) {
+        elem->position.x = elem->layer * HSPACING * width;
+        elem->position.h = height;
+        elem->position.w = width;
+
+        if (elements.size() == max_elements) {
+            elem->position.y = count * VSPACING * height;
+        }
+        else {
+            elem->position.y = step * (count + 1) + height * count;
+        }
+
+        count++;
+    }
+}
+
+void setConnections(Vertex* element, const std::vector<Vertex>& scheme, std::vector<Line>& connections) {
+    for(int& input : element->ins) {
+        Line connect(scheme[input], *element);
+        connections.push_back(connect);
+    }
+}
+
+void drawElementsAndConnections(SDL_Renderer* renderer, const logic_scheme& elements, const std::vector<Line>& connections) {
+    SDL_SetRenderDrawColor(renderer, 255,255,255, SDL_ALPHA_OPAQUE);
+
+    //Placing elements on the screen
+    for(const Vertex& element : elements.scheme) {
+        SDL_RenderDrawRectF(renderer, &element.position);
+    }
+
+    //Placing connection lines on the screen
+    for(const Line& connection_line : connections) {
+        SDL_RenderDrawLineF(renderer, connection_line.x1, connection_line.y1, connection_line.x2, connection_line.y2);
+    }
+}
+
+void drawBackground(SDL_Renderer* renderer) {
+    SDL_SetRenderDrawColor(renderer, 0,0,0, SDL_ALPHA_OPAQUE);
+    SDL_RenderClear(renderer);
+}
+
+int main(int argc, char* argv[]) {
+
+    std::cout << "Enter the file path: ";
+    std::string filename;
+    std::cin >> filename;
+
+    if(filename.empty()) {
+        std::cout << "Empty filename\n";
+        return EMPTY_FILENAME;
+    }
+
+    logic_scheme logicScheme;
+
+    bench_parser parser(logicScheme);
+    auto result = lorina::read_bench(filename, parser);
+    if(result == lorina::return_code::parse_error) {
+        std::cout << "Parser failure\n";
+        return PARSER_FAILURE;
+    }
+
+    logicScheme.queue_handling();
+    logicScheme.output(std::cout);
+
+    SDL_DisplayMode displayMode;
+    SDL_Window* window = nullptr;
+    SDL_Renderer* renderer = nullptr;
+    std::vector<Line> connections;
+
```

Comment:
Declare variables as close as possible to the point of use, init variables immediately with target values, e.g.
```
SDL_DisplayMode displayMode;
SDL_GetCurrentDisplayMode(0, &displayMode);
```